centinel/primitives/dnslib.py

```diff
@@ -171,36 +170,58 @@ def lookup_domain(self, domain, nameserver=None, log_prefix=''):
         sock.sendto(request.to_wire(), (nameserver, 53))
 
         # read the first response from the socket
-        reads, _, _ = select.select([sock], [], [], self.timeout)
-        # if we didn't get anything, then set the results to nothing
-        if len(reads) == 0:
+        try:
+            response = sock.recvfrom(4096)[0]
+            results['response1'] = b64encode(response)
+            resp = dns.message.from_wire(response)
+            results['response1-ips'] = parse_out_ips(resp)
+
+            # first domain name in response should be the same with query
+            # domain name
+            for entry in resp.answer:
+                if domain.lower() != entry.name.to_text().lower()[:-1]:
+                    logging.debug("%sWrong domain name %s for %s!"
+                                  % (log_prefix, entry.name.to_text().lower()[:-1], domain))
+                    results['response1-domain'] = entry.name.to_text().lower()[:-1]
```

Comment:
@lianke123321 are you storing the actual result with the wrong query domain name somewhere?